### ISSUE TITLE
I have a keras version of amsoftmax, but it not well in face converge is very slow

### ISSUE THREAD

**Issue by FlyEgle** [Joker316701882/Additive-Margin-Softmax#9]
![image](https://user-images.githubusercontent.com/26165705/38654705-460664fc-3e43-11e8-9dcb-e14a5d94b32e.png)

![image](https://user-images.githubusercontent.com/26165705/38654695-3d356a4e-3e43-11e8-8fdf-b239737ae13e.png)

I do not know where have  problem, can you have a look?

**Comment by vsatyakumar** [comment]
I suspect self.kernel needs to be the weight of the embedding layer (of the incoming layer to AMSoftmax). You can use model.layers[<layer_number>].get_weights() and assign it to self.kernel during the call.

**Comment by Joker316701882** [comment]
@FlyEgle 
Which layer are u using as face embeddings?
![image](https://user-images.githubusercontent.com/18679821/38793181-1ca87dca-418b-11e8-8de9-e353f22c0365.png)
Given this code, I assumed you didn't use "nomalized inputs", but original inputs as face embedding right? If so ,this is problem.

**Comment by FlyEgle** [comment]
A new keras amsoftmax below
![image](https://user-images.githubusercontent.com/26165705/38793426-d44f123e-4183-11e8-80bc-a083eb6ae78f.png)

This is the net-work cnn struct
![image](https://user-images.githubusercontent.com/26165705/38793400-b669c19c-4183-11e8-919f-931381313230.png)
Can you see if it is right?Beacuse i use m=0.35 and s = 30, i only can train 0.93 acc and val_acc is 0.84
in a small face datasets called FaceScrub which have 529 peoples

**Comment by Joker316701882** [comment]
@FlyEgle 
Logic seems no problem here. Evaluation code matters more. And to see whether your code is fine, simply train on larger dataset(CASIA, vggface,vggfacd2 etc...).  529 is too small. Reopen if needed.

**Comment by FlyEgle** [comment]
ok
